src/test/operators/join_test_runner.cpp

```diff
@@ -0,0 +1,530 @@
+        join_test_configuration.nullable_right = true;
+        join_test_configuration.predicate_condition = predicate_condition;
+
+        add_configuration_if_supported(join_test_configuration);
+      }
+    }
+
+    // The input tables are designed to have exclusive values. Test that these are handled correctly for different
+    // JoinModes by swapping the input tables.
+    // Additionally, go through all PredicateCondition to test especially the JoinSortMerge's different paths for these
+    for (const auto& predicate_condition : all_predicate_conditions) {
+      for (const auto& join_mode : all_join_modes) {
+        auto join_test_configuration = default_configuration;
+        join_test_configuration.join_mode = join_mode;
+        join_test_configuration.predicate_condition = predicate_condition;
+
+        join_test_configuration.swap_input_sides();
+
+        add_configuration_if_supported(join_test_configuration);
+      }
+    }
+
+    // Test all combinations of reference/data input tables. This tests mostly the composition of the output table
+    for (const auto& left_input_table_type : all_input_table_types) {
+      for (const auto& right_input_table_type : all_input_table_types) {
+        auto join_test_configuration = default_configuration;
+        join_test_configuration.input_left.table_type = left_input_table_type;
+        join_test_configuration.input_right.table_type = right_input_table_type;
+
+        add_configuration_if_supported(join_test_configuration);
+      }
+    }
+
+    // Test MPJ support for all join modes. Swap the input tables to trigger cases especially in the Anti* modes
+    // where a secondary predicate evaluating to FALSE might "save" a tuple from being discarded
+    for (const auto& join_mode : all_join_modes) {
+      for (const auto& secondary_predicates : all_secondary_predicate_sets) {
+        for (const auto& predicate_condition : {PredicateCondition::Equals, PredicateCondition::NotEquals}) {
+          for (const auto swap_input_sides : all_swap_input_sides) {
+            auto join_test_configuration = default_configuration;
+            join_test_configuration.join_mode = join_mode;
+            join_test_configuration.secondary_predicates = secondary_predicates;
+            join_test_configuration.predicate_condition = predicate_condition;
+
+            if (swap_input_sides) {
+              join_test_configuration.swap_input_sides();
+            }
+
+            add_configuration_if_supported(join_test_configuration);
+          }
+        }
+      }
+    }
+
+    // JoinHash specific configurations varying the number of radix_bits.
+    // The number of radix bits affects the partitioning inside the JoinHash operator, thus table/chunk sizes and
+    // different join modes need to be tested in combination, since they all use this partitioning.
+    if constexpr (std::is_same_v<JoinOperator, JoinHash>) {
+      for (const auto radix_bits : {1, 2, 5}) {
```

Comment:
Is `radix_bits = 0` tested as well somewhere?

Reply:
I probably assumed that was the default, but it is not. I added it here, let's see whether there are bugs ;)

Reply:
all seems fine